CPP_STYLE_GUIDE.md

```diff
@@ -2,28 +2,32 @@
 
 ## Table of Contents
 
-* [Left-leaning (C++ style) asterisks for pointer declarations](#left-leaning-c-style-asterisks-for-pointer-declarations)
-* [2 spaces of indentation for blocks or bodies of conditionals](#2-spaces-of-indentation-for-blocks-or-bodies-of-conditionals)
-* [4 spaces of indentation for statement continuations](#4-spaces-of-indentation-for-statement-continuations)
-* [Align function arguments vertically](#align-function-arguments-vertically)
-* [Initialization lists](#initialization-lists)
-* [CamelCase for methods, functions and classes](#camelcase-for-methods-functions-and-classes)
-* [snake\_case for local variables and parameters](#snake_case-for-local-variables-and-parameters)
-* [snake\_case\_ for private class fields](#snake_case_-for-private-class-fields)
-* [Space after `template`](#space-after-template)
+* [Formatting](#formatting)
+  * [Left-leaning (C++ style) asterisks for pointer declarations](#left-leaning-c-style-asterisks-for-pointer-declarations)
+  * [2 spaces of indentation for blocks or bodies of conditionals](#2-spaces-of-indentation-for-blocks-or-bodies-of-conditionals)
+  * [4 spaces of indentation for statement continuations](#4-spaces-of-indentation-for-statement-continuations)
+  * [Align function arguments vertically](#align-function-arguments-vertically)
+  * [Initialization lists](#initialization-lists)
+  * [CamelCase for methods, functions and classes](#camelcase-for-methods-functions-and-classes)
+  * [snake\_case for local variables and parameters](#snake_case-for-local-variables-and-parameters)
+  * [snake\_case\_ for private class fields](#snake_case_-for-private-class-fields)
+  * [Space after `template`](#space-after-template)
+* [Memory Management](#memory-management)
+  * [Memory allocation](#memory-allocation)
+  * [Use `nullptr` instead of `NULL` or `0`](#use-nullptr-instead-of-null-or-0)
+  * [Ownership and Smart Pointers](#ownership-and-smart-pointers)
 * [Type casting](#type-casting)
-* [Memory allocation](#memory-allocation)
```

Comment:
Can you put "Others" as a top level item and put other rules under it?

Reply:
Done